crates/pink/sidevm/macro/src/tests.rs

```diff
@@ -0,0 +1,29 @@
+#[test]
+fn test_ocall() {
+    let stream = crate::macro_ocall::patch(syn::parse_quote! {
+        pub trait Ocall {
```

Comment:
In the initial design, `fast_input` and `fast_return` is for those primitive types can be lost-less-ly cast into i32 (i8,u8,i16,u16, i32, u32).
Later, I extend the acceptable arguments with a pair trait `ArgEncode/ArgDecode` which can support some small fixed-size types can be encoded into the i32 arguments. For example, `&[u8]` and `&str` are encoded into two arguments (ptr + len) via fast_input.

For non-fast args/ret, it use scale codec. So, it can accept all types implemented scale codec. The overhead is:
- Encode/decode itself would add an extra se/derilization.
- May allocate dynamic memory. (Small types are optimized with `TinyVec` to avoid allocation).
- The return type is dynamic length, So it need an extra ocall the fetch the result.